bootstrap/bootstrap/src/hub.rs

```diff
@@ -34,53 +43,101 @@ impl Repo {
         T: AsRef<str>,
     {
         Repo {
-            base: String::from(HUB),
             name: String::from(name.as_ref()),
         }
     }
 
     pub fn list(&self) -> Result<Vec<Flist>> {
-        let url = format!("{}/api/flist/{}", self.base, self.name,);
+        let url = format!("{}/api/flist/{}", HUB, self.name,);
 
         let response = get(&url)?;
         let mut info: Vec<Flist> = match response.status() {
             StatusCode::OK => response.json()?,
             s => bail!("failed to get flist info: {}", s),
         };
         for flist in info.iter_mut() {
-            flist.url = format!("{}/{}/{}", self.base, self.name, flist.name);
+            flist.url = format!("{}/{}/{}", HUB, self.name, flist.name);
         }
 
         Ok(info)
     }
 
+    pub fn list_tag<S: AsRef<str>>(&self, tag: S) -> Result<Option<Vec<Flist>>> {
+        let tag = tag.as_ref();
+
+        let url = format!("{}/api/flist/{}/tags/{}", HUB, self.name, tag);
+        let response = get(&url)?;
+        let mut info: Vec<Flist> = match response.status() {
+            StatusCode::OK => response.json()?,
+            StatusCode::NOT_FOUND => return Ok(None),
+            s => bail!("failed to get flist info: {}", s),
+        };
+
+        // when listing tags. The flists inside have target as full name
+        // so we need
+        for flist in info.iter_mut() {
+            if flist.kind == Kind::Regular {
+                // this is impossible because tags can only have symlinks
+                continue;
+            }
+
+            let target = match &flist.target {
+                None => {
+                    // this is also not possible may be we should return an error
+                    // but we can just skip for now
+                    continue;
+                }
+                Some(target) => target,
+            };
+
+            flist.url = format!("{}/{}", HUB, target);
+        }
+
+        Ok(Some(info))
+    }
+
+    /// gets flist information from name. the flist must be of type flist or symlink
     pub fn get<T>(&self, flist: T) -> Result<Flist>
     where
         T: AsRef<str>,
     {
-        let url = format!(
-            "{}/api/flist/{}/{}/light",
-            self.base,
-            self.name,
-            flist.as_ref()
-        );
+        let url = format!("{}/api/flist/{}/{}/light", HUB, self.name, flist.as_ref());
 
         let response = get(&url)?;
         let mut info: Flist = match response.status() {
             StatusCode::OK => response.json()?,
             s => bail!("failed to get flist info: {}", s),
         };
-        info.url = format!("{}/{}/{}", self.base, self.name, flist.as_ref());
+        info.url = format!("{}/{}/{}", HUB, self.name, flist.as_ref());
         Ok(info)
     }
 }
 
 impl Flist {
+    /// tag_link return the target repo and tag name for
```

Comment:
should we add in the doc that this method panics if the kind is not `TagLink`?

Reply:
or should we not panic? :D

Reply:
I will add that to the docs that it will panic. The panic is okay in this case because this means u have a development error. (there is no way u can call this with wrong type unless u have wrong logic) hence the panic